pkg/csi/csi.go

```diff
@@ -58,19 +59,37 @@ func InitializeSidecars(ver string) error {
 		return err
 	}
 
-	sv := version.MustParseGeneric(ver)
-
-	for _, image := range si {
-		v := version.MustParseGeneric(image.Version)
-		if sv.Major() == v.Major() && sv.Minor() == v.Minor() {
-			sidecarImages = &image
-			break
+	pltVersion := version.MustParseGeneric(ver)
+
+	// this helps in not parsing the version twice in loop.
+	closestVersion := pltVersion.WithMinor(0)
+	for idx := range si {
+		siVersion := version.MustParseGeneric(si[idx].Version)
+		log.Info("verifying sidecar images version for best compatibility to platform",
+			"sidecar", siVersion, "platform", pltVersion)
+
+		// only check sidecar image versions that are not higher than platform
+		if siVersion.Major() == pltVersion.Major() && siVersion.Minor() <= pltVersion.Minor() {
+			// filter sidecar closest to platform version
+			if closestVersion.Minor() <= siVersion.Minor() {
+				sidecarImages = &si[idx]
+				closestVersion = siVersion
+			}
+			if closestVersion.Minor() == pltVersion.Minor() { // exact match and early exit
+				break
+			}
+		} else {
+			log.Info("skipping sidecar images version greater than platform version",
+				"sidecar", siVersion, "platform", pltVersion)
```

Comment:
You already logged the versions on the last message in the log. No need to log again.
```suggestion
			log.Info("skipping sidecar images: version greater than platform") 
```